go-controller/pkg/ovn/gateway.go

```diff
@@ -403,7 +403,7 @@ func (gw *GatewayManager) GatewayInit(
 			types.NetworkExternalID:  gw.netInfo.GetNetworkName(),
 			types.TopologyExternalID: gw.netInfo.TopologyType(),
 		}
-		if gw.netInfo.IsPrimaryNetwork() && gw.netInfo.TopologyType() == types.Layer2Topology {
+		if gw.netInfo.IsPrimaryNetwork() && gw.netInfo.TopologyType() == types.Layer2Topology && config.IPv6Mode {
```

Comment:
What about single stack UDNs created on dualstack clusters?
Should we rely on `gw.netInfo.IPMode()` instead?

Reply:
even if its a single stack UDN on a dualstack cluster I'd think we need the v6 options set cause the network is dualstack right? A user should be able to add a v6 subnet as a day2 config of converting the network from singlestack to dualstack if cluster supports it right? OR is that not a thing ? - I regret I thought about this.. so how exactly does dualstack cluster conversion work ? if we had older pods... would the restart create v6IPs for existing pods ?

Reply:
> even if its a single stack UDN on a dualstack cluster I'd think we need the v6 options set cause the network is dualstack right? A user should be able to add a v6 subnet as a day2 config of converting the network from singlestack to dualstack if cluster supports it right? OR is that not a thing ? 

I disagree. The user defined the network as single-stack. And the UDN is immutable. 

What the user can do is create *another* dual-stack immutable UDN egressing to the dual-stack network.

I agree w/ @kyrtapz that what's important is the `NetInfo` specification - as long as the underlying network support the user's intents - i.e. single stack UDN on a dual-stack cluster is OK; dual-stack UDN on a single stack cluster is **not** OK.

Reply:
I'll check both, that the UDN is ipv6 and ipv6 is allowed globally in ovnk. Does that work?

Reply:
Makes sense to me.

Reply:
hmm what I meant was is there any harm in setting these specific configs even if the NAD is singlestack....... but yeah taking both configs into account is the best probably needs a util in itself to be reusbale